### PR TITLE
add compaction sleep interval

### PR DESCRIPTION
**Pull request by AlexStocks**
Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

### CODE REVIEW

File: server/config/config.go [hunk 1]
```diff
@@ -114,6 +114,7 @@ type ServerConfig struct {
 	AutoCompactionRetention time.Duration
 	AutoCompactionMode      string
 	CompactionBatchLimit    int
+	CompactionSleepInterval int
```
Comment (ptabor) [inline server/config/config.go:117]: time.Duration ?

File: server/embed/config.go [hunk 2]
```diff
@@ -317,6 +317,7 @@ type Config struct {
 	// ExperimentalEnableLeaseCheckpoint enables primary lessor to persist lease remainingTTL to prevent indefinite auto-renewal of long lived leases.
 	ExperimentalEnableLeaseCheckpoint       bool          `json:"experimental-enable-lease-checkpoint"`
 	ExperimentalCompactionBatchLimit        int           `json:"experimental-compaction-batch-limit"`
+	ExperimentalCompactionSleepInterval     int           `json:"experimental-compaction-sleep-interval"`
```
Comment (ptabor) [inline server/embed/config.go:320]: time.Duration ?

Please add a comment describing what is it.

File: server/etcdmain/config.go [hunk 3]
```diff
@@ -282,6 +282,7 @@ func newConfig() *config {
 
 	fs.BoolVar(&cfg.ec.ExperimentalEnableLeaseCheckpoint, "experimental-enable-lease-checkpoint", false, "Enable to persist lease remaining TTL to prevent indefinite auto-renewal of long lived leases.")
 	fs.IntVar(&cfg.ec.ExperimentalCompactionBatchLimit, "experimental-compaction-batch-limit", cfg.ec.ExperimentalCompactionBatchLimit, "Sets the maximum revisions deleted in each compaction batch.")
+	fs.IntVar(&cfg.ec.ExperimentalCompactionSleepInterval, "experimental-compaction-sleep-interval", cfg.ec.ExperimentalCompactionSleepInterval, "Sets the sleep interval in each compaction batch.")
```
Comment (ptabor) [inline server/etcdmain/config.go:285]: in -> between ?

File: server/mvcc/kvstore.go [hunk 4]
```diff
@@ -52,9 +52,11 @@ const (
 
 var restoreChunkKeys = 10000 // non-const for testing
 var defaultCompactBatchLimit = 1000
+var minimumBatchInterval = 10 // 10ms
```
Comment (ptabor) [inline server/mvcc/kvstore.go:55]: time.Duration(10*time.millis)

**Review by ptabor — CHANGES_REQUESTED** [review]
1. Please describe in the PR definition what's the motivation for the change and in what circumstances user's should use the higher values. 
2. Please add an information to the CHANGELOG file

File: server/config/config.go (hunk 1 above)
Comment (AlexStocks) [reply server/config/config.go:117]: done

File: server/mvcc/kvstore.go (hunk 4 above)
Comment (AlexStocks) [reply server/mvcc/kvstore.go:55]: done

File: server/etcdmain/config.go (hunk 3 above)
Comment (AlexStocks) [reply server/etcdmain/config.go:285]: done

File: server/embed/config.go (hunk 2 above)
Comment (AlexStocks) [reply server/embed/config.go:320]: done

**Comment by AlexStocks** [comment]
@ptabor sorry, there are many conflicts which are difficult to fix. I submit another pr https://github.com/etcd-io/etcd/pull/13018 instead.

**Pull request closed without merging**